osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/WatchScoresCommand.cs

```diff
@@ -0,0 +1,182 @@
+                Console.WriteLine($"Pushing imported scores to redis queue {scoreStatisticsQueueProcessor.QueueName}");
+            }
+
+            if (DryRun)
+                Console.WriteLine("RUNNING IN DRY RUN MODE.");
+
+            Console.WriteLine();
+            Console.WriteLine("Starting processing in 5 seconds...");
+            await Task.Delay(5000, cancellationToken);
+
+            using (var dbMainQuery = DatabaseAccess.GetConnection())
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    HighScore[] scores = (await dbMainQuery.QueryAsync<HighScore>(
+                            $"SELECT * FROM {scoreTable} "
+                            // A delay is applied here to be sure that high_score_id has been populated.
+                            // This is a safety as web-10 doesn't hold a transaction between the insert to `osu_scores` and `osu_scores_high`.
+                            // Worst case scenario is that a high score exists twice, once with missing `legacy_score_id`.
+                            + "WHERE score_id >= @lastScoreId AND date < DATE_SUB(NOW(), INTERVAL 5 SECOND) AND high_score_id IS NULL ORDER BY score_id LIMIT 50", new
+                            {
+                                lastScoreId = lastScoreId,
+                                RulesetId,
+                            }))
+                        .ToArray();
+
+                    if (scores.Length == 0)
+                    {
+                        Thread.Sleep(500);
+                        continue;
+                    }
+
+                    // Need to obtain score_id before zeroing them out.
+                    lastScoreId = scores.Last().score_id;
+
+                    var inserter = new BatchInserter(ruleset, scores, importLegacyPP: SkipScoreProcessor, dryRun: DryRun);
+
+                    while (!inserter.Task.IsCompleted)
+                    {
+                        outputProgress();
+                        Thread.Sleep(10);
+                    }
+
+                    if (inserter.Task.IsFaulted)
+                    {
+                        Console.WriteLine("ERROR: Inserter failed. Aborting for safety.");
+                        Console.WriteLine($"Running batches were processing up to {lastScoreId}.");
+                        Console.WriteLine();
+
+                        throw inserter.Task.Exception!;
+                    }
+
+                    pushCompletedScoreToQueue(inserter);
+
+                    Console.WriteLine($"Workers processed up to (score_id: {lastScoreId})");
+                    lastScoreId++;
+                }
+            }
+
+            outputFinalStats();
+            return 0;
+        }
+
+        private void pushCompletedScoreToQueue(BatchInserter inserter)
+        {
+            if (scoreStatisticsQueueProcessor != null)
+            {
+                Debug.Assert(inserter.ElasticScoreItems.Any());
```

Comment:
I'm very confused here. The check is on `scoreStatisticsQueueProcessor`, but this assert is on `ElasticScoreItems`. First, I would expect this assert to be _the other way around_, i.e.

```suggestion
                Debug.Assert(!inserter.ElasticScoreItems.Any());
```

because either `ElasticScoreItems` or `ScoreStatisticsItems` should be empty, and this is the branch pushing the second of these. But, more importantly, why does anything in here check if there are any items in `ElasticScoreItems`, _if nothing in this command actually pushes anything to ES ever, and arguably should never do so?_

I won't mention the fact that `SkipScoreProcessor` is a rather weird flag in this command, despite the fact that it _kinda_ does what you'd expect it to, but it does so in a very weird way (by appending items to `ElasticScoreItems` which will never be pushed to ES).

Reply:
ES indexing is never a thing with non-high scores. I've removed this remaining reference.